onnxruntime/contrib_ops/cuda/layer_norm_impl.cu

```diff
@@ -355,19 +356,66 @@ void HostApplyLayerNorm(
   const uint64_t maxGridY = prop.maxGridSize[1];
   const int warp_size = prop.warpSize;
   ORT_ENFORCE(warp_size == GPU_WARP_SIZE);
+  
+  bool custom_supported_dim = std::find(std::begin(SUPPORTED_REDUCTION_DIM), std::end(SUPPORTED_REDUCTION_DIM), n2) != std::end(SUPPORTED_REDUCTION_DIM);
+  bool supported_half2 = (prop.major >= 7) || !(std::is_same<T, half>::value);
+  if(custom_supported_dim && supported_half2){
+    launch_custom_layer_norm<T,U>(
+      output,
+      input,
+      gamma,
+      beta,
+      U(epsilon),
+      n1,
+      n2,
+      invvar,
+      mean);
+  }
+  else{
+    const dim3 threads(warp_size, 4, 1);
+    const dim3 blocks(1, std::min((uint64_t)n1, maxGridY), 1);
+    int nshared =
+        threads.y > 1 ? threads.y * sizeof(U) + (threads.y / 2) * sizeof(U) : 0;
+    cuApplyLayerNorm<<<blocks, threads, nshared, 0>>>(
+        output,
+        mean,
+        invvar,
+        input,
+        n1, n2,
+        U(epsilon),
+        gamma, beta);
+  }
+}
 
+template <>
+void HostApplyLayerNorm(
+    const cudaDeviceProp& prop,
+    double* output,
+    double* mean,
+    double* invvar,
+    const double* input,
+    int64_t n1,
+    int64_t n2,
+    double epsilon,
+    const double* gamma,
+    const double* beta) {
+  const uint64_t maxGridY = prop.maxGridSize[1];
+  const int warp_size = prop.warpSize;
+  ORT_ENFORCE(warp_size == GPU_WARP_SIZE);
+  
   const dim3 threads(warp_size, 4, 1);
   const dim3 blocks(1, std::min((uint64_t)n1, maxGridY), 1);
   int nshared =
-      threads.y > 1 ? threads.y * sizeof(U) + (threads.y / 2) * sizeof(U) : 0;
+      threads.y > 1 ? threads.y * sizeof(double) + (threads.y / 2) * sizeof(double) : 0;
```

Comment:
why does it need to be specified for double?

Reply:
The windows compiler couldn't automatically instantiate the template for double previously, causing the windows CI to fail. Had to explicitly call the old kernel for double.